MPSGameLift/Code/Source/MPSGameLiftClientSystemComponent.cpp

```diff
@@ -68,6 +70,19 @@ namespace MPSGameLift
         request.m_sessionId = gameSessionId;
         request.m_playerId = playerId.ToString<AZStd::string>();
 
+        // Configure the GameLift client with the proper region; 
+        // Note: fallback region is defined inside default_aws_resource_mappings.json
+        AZStd::string region;
+        const uint32_t regionIndex = 3; // Region is the 4th value in a GameLift Arn. arn:aws:gamelift:<region>
+        AZStd::vector<AZStd::string> tokenizedGameLiftArn;
+        AZ::StringFunc::Tokenize(gameSessionId, tokenizedGameLiftArn, ":");
+        if (tokenizedGameLiftArn.size() >= regionIndex)
+        {
+            region = tokenizedGameLiftArn[regionIndex];
+        }
```

Comment:
nit: repeated code, could it be turned into a function?

Reply:
https://github.com/o3de/o3de/pull/16158